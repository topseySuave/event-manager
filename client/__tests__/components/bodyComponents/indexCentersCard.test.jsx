import React from 'react';
import { shallow } from 'enzyme';
import { IndexCenterCardHolder, mapStateToProps, mapDispatchToProps } from
  '../../../src/components/bodyComponents/IndexCenterCardHolder';
import {
  fetchCentersAction,
  loadMoreCenters
} from '../../../src/actions/center-actions/fetchCenterAction';

let wrapper, instance, mounted;
let store, componentProps;

const getComponent = (props) => {
  mounted = shallow(<IndexCenterCardHolder {...props} />);
  return mounted;
};

let LoadMoreSpy = jest.spyOn(IndexCenterCardHolder.prototype, 'loadMore');

describe('IndexCenterCardHolder component', () => {
  test('should mount', () => {
    componentProps = {
      fetchCentersAction,
      centerStore: {
        centers: [],
      },
      loadMoreCenters
    };
    wrapper = getComponent(componentProps);
    expect(wrapper).toMatchSnapshot();
  });

  test('should update when new properties a provided', () => {
    componentProps.centerStore = {
      centers: [],
      loadingmore: true,
      loadmore: true,
      meta: {
        page: 2,
        pageCount: 2,
        pageSize: 5,
        totalCount: 6
      }
    };
    wrapper = getComponent(componentProps);
    wrapper.setProps(componentProps);
    expect(wrapper.state('isLoading')).toBeFalsy();
  });

  test('should show the loadmore button', () => {
    wrapper = getComponent(componentProps);
    wrapper.setState({
      isLoading: false,
      pageCount: 2
    });
    wrapper.instance().showLoadMoreButton();
  });

  test('should be able to loadmore', () => {
    wrapper = getComponent(componentProps);
    mapDispatchToProps(jest.fn());
    mapStateToProps({});
    wrapper.instance().loadMore();
    expect(LoadMoreSpy).toBeCalled();
  });
});
