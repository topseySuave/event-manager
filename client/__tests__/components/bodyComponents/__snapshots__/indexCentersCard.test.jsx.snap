// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexCenterCardHolder component IndexCenterCardHolder component should mount 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IndexCenterCardHolder
    centerStore={
      Object {
        "centers": Array [],
      }
    }
    fetchCentersAction={[Function]}
    loadMoreCenters={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container popular__events"
      >
        <div
          style={
            Object {
              "height": "500px",
              "marginTop": "100px",
            }
          }
        >
          <CircularLoader />
        </div>
      </div>,
      "className": "popular__events_holdr",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          style={
            Object {
              "height": "500px",
              "marginTop": "100px",
            }
          }
        >
          <CircularLoader />
        </div>,
        "className": "container popular__events",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <CircularLoader />,
          "style": Object {
            "height": "500px",
            "marginTop": "100px",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container popular__events"
        >
          <div
            style={
              Object {
                "height": "500px",
                "marginTop": "100px",
              }
            }
          >
            <CircularLoader />
          </div>
        </div>,
        "className": "popular__events_holdr",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            style={
              Object {
                "height": "500px",
                "marginTop": "100px",
              }
            }
          >
            <CircularLoader />
          </div>,
          "className": "container popular__events",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <CircularLoader />,
            "style": Object {
              "height": "500px",
              "marginTop": "100px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
